sycl/include/CL/sycl/accessor2.hpp

```diff
@@ -332,17 +332,15 @@ class accessor :
 
 public:
   using value_type = DataT;
-  using reference = typename detail::PtrValueType<DataT, AS>::type &;
-  using const_reference = const reference;
+  using reference = DataT &;
+  using const_reference = const DataT &;
```

Comment:
Just out of curiosity: is there any problem with old definition of `const_reference` or you just prefer this style?

Reply:
It is not stylistic fix. The current version changes the type on DEVICE, e.g. __global T& to T&.
CTS tests reported incompatibility of types (they have asserts for accessor::reference and accessor::const_reference).

Reply:
My question is about `const_reference` only.
```c++
using reference = DataT &;
using const_reference = const reference;
```
Does this version has the problem you described?

Reply:
I tried that initially. It surprisingly gave me an error. I'll post the error here later.

Reply:
Fixed. 
The error that I observed previously with 'using const_reference = const reference;' was caused by the fact that I edited the file on Linux machine using Windows app which set the date/time of modification incorrectly, thus make/build did not re-built accessor2.hpp that time.